<a href="https://colab.research.google.com/github/JingYannn/CatastrophicForgetting-BERT/blob/master/eval_prediction_comparation.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import csv
import numpy as np
from collections import defaultdict

In [ ]:
def read_data_from_file(data_dir, quotechar=None):
  with open(data_dir, "r", encoding="utf-8-sig") as f:
    return list(csv.reader(f, delimiter='\t', quotechar=quotechar))

In [ ]:
eval_results1 = read_data_from_file('eval_results_cola1.txt')
print(eval_results1)
print(len(eval_results1))

[['index', 'label', 'prediction'], ['906', '0', '0'], ['501', '0', '0'], ['958', '1', '1'], ['443', '1', '1'], ['755', '0', '1'], ['513', '1', '1'], ['758', '1', '1'], ['451', '0', '0'], ['71', '1', '1'], ['994', '1', '1'], ['825', '0', '1'], ['331', '1', '1'], ['976', '1', '1'], ['1013', '0', '0'], ['27', '1', '1'], ['943', '1', '1'], ['528', '1', '1'], ['922', '0', '0'], ['980', '0', '1'], ['780', '1', '1'], ['1023', '1', '1'], ['1011', '0', '1'], ['312', '1', '1'], ['957', '1', '1'], ['971', '1', '1'], ['437', '0', '0'], ['309', '1', '1'], ['631', '0', '1'], ['944', '1', '1'], ['106', '1', '1'], ['963', '0', '0'], ['813', '0', '0'], ['134', '1', '1'], ['53', '1', '1'], ['262', '1', '1'], ['791', '1', '1'], ['514', '1', '1'], ['225', '0', '0'], ['702', '0', '0'], ['936', '1', '1'], ['230', '0', '0'], ['207', '1', '1'], ['216', '1', '1'], ['367', '0', '0'], ['868', '0', '0'], ['607', '1', '1'], ['420', '1', '1'], ['85', '1', '0'], ['158', '1', '1'], ['811', '1', '1'], ['638', '1', '1'

In [ ]:
dict_eval_results = defaultdict()
for i in range(len(eval_results1)):
  dict_eval_results[eval_results1[i][0]] = [eval_results1[i][1], eval_results1[i][2]]
print(dict_eval_results)

defaultdict(None, {'index': ['label', 'prediction'], '906': ['0', '0'], '501': ['0', '0'], '958': ['1', '1'], '443': ['1', '1'], '755': ['0', '1'], '513': ['1', '1'], '758': ['1', '1'], '451': ['0', '0'], '71': ['1', '1'], '994': ['1', '1'], '825': ['0', '1'], '331': ['1', '1'], '976': ['1', '1'], '1013': ['0', '0'], '27': ['1', '1'], '943': ['1', '1'], '528': ['1', '1'], '922': ['0', '0'], '980': ['0', '1'], '780': ['1', '1'], '1023': ['1', '1'], '1011': ['0', '1'], '312': ['1', '1'], '957': ['1', '1'], '971': ['1', '1'], '437': ['0', '0'], '309': ['1', '1'], '631': ['0', '1'], '944': ['1', '1'], '106': ['1', '1'], '963': ['0', '0'], '813': ['0', '0'], '134': ['1', '1'], '53': ['1', '1'], '262': ['1', '1'], '791': ['1', '1'], '514': ['1', '1'], '225': ['0', '0'], '702': ['0', '0'], '936': ['1', '1'], '230': ['0', '0'], '207': ['1', '1'], '216': ['1', '1'], '367': ['0', '0'], '868': ['0', '0'], '607': ['1', '1'], '420': ['1', '1'], '85': ['1', '0'], '158': ['1', '1'], '811': ['1', '1']

In [ ]:
eval_results2 = read_data_from_file('eval_results_cola2.txt')
print(eval_results2)
print(len(eval_results2))

[['index', 'label', 'prediction'], ['753', '1', '1'], ['772', '1', '1'], ['79', '1', '1'], ['1025', '1', '1'], ['686', '1', '1'], ['427', '1', '1'], ['797', '1', '1'], ['36', '1', '1'], ['637', '0', '1'], ['822', '1', '1'], ['846', '1', '1'], ['914', '1', '1'], ['247', '1', '1'], ['823', '0', '1'], ['428', '0', '1'], ['361', '1', '1'], ['1017', '0', '1'], ['260', '1', '1'], ['328', '0', '1'], ['649', '0', '1'], ['1005', '1', '1'], ['542', '0', '1'], ['1007', '1', '1'], ['818', '0', '1'], ['323', '1', '1'], ['828', '0', '1'], ['450', '1', '1'], ['32', '0', '1'], ['117', '0', '1'], ['459', '1', '1'], ['130', '1', '1'], ['222', '1', '1'], ['852', '0', '1'], ['1038', '1', '1'], ['452', '0', '1'], ['523', '1', '1'], ['936', '1', '1'], ['157', '1', '1'], ['950', '1', '1'], ['929', '1', '1'], ['170', '0', '0'], ['216', '1', '1'], ['169', '1', '1'], ['581', '1', '1'], ['280', '0', '1'], ['827', '0', '1'], ['613', '0', '0'], ['249', '1', '1'], ['677', '1', '1'], ['612', '1', '1'], ['70', '1', '

In [ ]:
for i in range(len(eval_results2)):
  assert dict_eval_results[eval_results2[i][0]][0] == eval_results2[i][1]
  dict_eval_results[eval_results2[i][0]].append(eval_results2[i][2])
print(len(dict_eval_results))

1044


In [ ]:
dict_eval_results.pop('index')

['label', 'prediction', 'prediction']

In [ ]:
print(len(dict_eval_results))

1043


In [ ]:
lists = sorted(dict_eval_results.items(), key=lambda d:int(d[0]))

print(lists)

[('0', ['1', '1', '1']), ('1', ['1', '1', '1']), ('2', ['1', '1', '1']), ('3', ['1', '1', '1']), ('4', ['0', '1', '1']), ('5', ['0', '1', '1']), ('6', ['0', '0', '1']), ('7', ['1', '1', '1']), ('8', ['1', '1', '1']), ('9', ['1', '1', '1']), ('10', ['1', '1', '1']), ('11', ['1', '1', '1']), ('12', ['1', '1', '1']), ('13', ['0', '1', '1']), ('14', ['1', '1', '1']), ('15', ['1', '1', '1']), ('16', ['0', '1', '1']), ('17', ['1', '1', '1']), ('18', ['1', '1', '1']), ('19', ['1', '1', '1']), ('20', ['1', '1', '1']), ('21', ['1', '1', '1']), ('22', ['1', '1', '1']), ('23', ['0', '0', '1']), ('24', ['1', '1', '1']), ('25', ['1', '1', '1']), ('26', ['1', '1', '1']), ('27', ['1', '1', '1']), ('28', ['0', '0', '1']), ('29', ['0', '1', '1']), ('30', ['1', '1', '1']), ('31', ['1', '1', '1']), ('32', ['0', '1', '1']), ('33', ['0', '1', '1']), ('34', ['1', '0', '1']), ('35', ['1', '1', '1']), ('36', ['1', '1', '1']), ('37', ['1', '1', '1']), ('38', ['1', '1', '1']), ('39', ['0', '1', '1']), ('40', ['

In [ ]:
print(lists[1][0])

1


### CoLA -> MRPC -> NER

### CoLA: 0(ungrammatical), 1(grammatical)



**cnt1** : indexes predicted correctly in the first time => 860

**cnt2** : indexes predicted correctly in the second time => 789

**cnt3** : indexes predicted correctly in the first time but predicted uncorrectly in the second time (forget) => 118
  - **cnt3_0** : how many label '0' are forgot? => 117
  - **cnt3_1** : how many label '1' are forgot? => 1

**cnt4** : indexes predicted correctly in the second time but predicted uncorrectly in the first time => 47
  - **cnt4_0** : how many label '0' are learned? => 0
  - **cnt4_1** : how many label '1' are learned? => 47

In [ ]:
def accuracy(lists):
  cnt1= []
  cnt2 = []
  cnt3 = []
  cnt3_0 = []
  cnt3_1 = []
  cnt4 = []
  cnt4_0 = []
  cnt4_1 = []
  for i, seq in enumerate(lists):
    if lists[i][1][0] == lists[i][1][1]:
      cnt1.append(i)
    if lists[i][1][0] == lists[i][1][2]:
      cnt2.append(i)
  
  for i in cnt1:
    if i not in cnt2:
      cnt3.append(i)
      if lists[i][1][0] == '0':
        cnt3_0.append(i)
      elif lists[i][1][0] == '1':
        cnt3_1.append(i)

  for i in cnt2:
    if i not in cnt1:
      cnt4.append(i)
      if lists[i][1][0] == '0':
        cnt4_0.append(i)
      elif lists[i][1][0] == '1':
        cnt4_1.append(i)
  return len(cnt1), len(cnt2), len(cnt3), len(cnt3_0),len(cnt3_1), len(cnt4), len(cnt4_0),len(cnt4_1)
accuracy(lists)

(864, 772, 139, 119, 20, 47, 11, 36)

In [ ]:
line = '1000'+'\t'+'apple'+'\t'+'pear\n'
splits = line.split('\t')
print(splits)
splits[0].isdigit()
splits[2][:-1]

['1000', 'apple', 'pear\n']


'pear'

In [ ]:
def read_data_from_file_ner(data_dir):
  dic = defaultdict()
  labels = []
  preds = []
  with open(data_dir, encoding='utf-8') as f:
    for i,line in enumerate(f):
      # print(line)
      if i == 0:
        continue
      if line[0] == '\n':
        if len(labels) > 0:
          # print(key)
          # print(labels)
          # print(preds)
          dic[key] = [labels,preds]
          labels = []
          preds = []
      else:
        splits = line.split('\t')
        # print(splits)
        if splits[0].isdigit and splits[0] != '':
          key = splits[0]
        labels.append(splits[1])
        preds.append(splits[2][:-1])
  return dic


In [ ]:
dic1 = read_data_from_file_ner('out/eval_results_ner1.txt')
dic2 = read_data_from_file_ner('out/eval_results_ner2.txt')

list1 = sorted(dic1.items(), key=lambda x:int(x[0]))
list2 = sorted(dic2.items(), key=lambda x:int(x[0]))
print(len(list1))
print(len(list2))
print(list2[-1])

3250
3250
('3250', [['O', 'B-ORG', 'I-ORG', 'O'], ['O', 'B-LOC', 'O', 'O']])


In [ ]:
cpm_dict = defaultdict()
for i, (item1,item2) in enumerate(zip(list1, list2)):
  assert item1[0] == item2[0]
  assert len(item1[1][0]) == len(item2[1][0])
  cpm_dict[item1[0]] = [item1[1][0], item1[1][1], item2[1][1]]

In [ ]:
label_list = ["O", "B-MISC", "I-MISC", "B-PER", "I-PER", "B-ORG", "I-ORG", "B-LOC", "I-LOC"]

**cnt1** : how many times is each label correctly predicted in the first time? 

**cnt2** : how many times is each label correctly predicted in the second time? 

**cnt3** : how many times is each label correctly predicted in the first time but uncorrectly in the second time? 

**cnt4** : how many times is each label correctly predicted in the second time but uncorrectly in the first time? 

In [ ]:
cnt1 = [0]*len(label_list)
cnt2 = [0]*len(label_list)
cnt3 = [0]*len(label_list)
cnt4 = [0]*len(label_list)
for item in cpm_dict.items():
  # print(item)
  for l1,l2,l3 in zip(item[1][0], item[1][1], item[1][2]):
    # print(l1,l2,l3)
    if l1 == l2:
      for i,label in enumerate(label_list):
        if l1 == label:
          cnt1[i] += 1
    if l1 == l3:
      for i,label in enumerate(label_list):
        if l1 == label:
          cnt2[i] += 1 
    if l1 == l2 and l1 != l3:
      for i,label in enumerate(label_list):
        if l1 == label:
          cnt3[i] += 1  
    if l1 == l3 and l1 != l2:
      for i,label in enumerate(label_list):
        if l1 == label:
          cnt4[i] += 1    

print(cnt1)  
print(cnt2) 
print(cnt3) 
print(cnt4) 
label_list = ["O", "B-MISC", "I-MISC", "B-PER", "I-PER", "B-ORG", "I-ORG", "B-LOC", "I-LOC"]
ori_cnt = [42746, 922, 346, 1839, 1304, 1341, 751, 1837, 257]
perc = [round(i2/i1, 4) for (i1,i2) in zip(ori_cnt,cnt3)]
print(perc)

[42587, 846, 291, 1814, 1295, 1260, 679, 1789, 237]
[42436, 850, 285, 1611, 1282, 1201, 684, 1665, 227]
[182, 17, 16, 203, 14, 66, 21, 130, 11]
[31, 21, 10, 0, 1, 7, 26, 6, 1]
[0.0043, 0.0184, 0.0462, 0.1104, 0.0107, 0.0492, 0.028, 0.0708, 0.0428]


In [ ]:
a = [42746, 922, 346, 1839, 1304, 1341, 751, 1837, 257]
b = [182, 17, 16, 203, 14, 66, 21, 130, 11]
res = [round(i2/i1, 4) for (i1,i2) in zip(a,b)]
print(res)

[0.0043, 0.0184, 0.0462, 0.1104, 0.0107, 0.0492, 0.028, 0.0708, 0.0428]


NER CoLA

In [ ]:
!bash run_cl.sh

2020-12-12 12:51:04.100339: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
Namespace(cuda=True, data_dir='data', device=device(type='cuda'), do_lower_case=True, eval_batch_size=32, eval_during_training=False, max_seq_length=128, model_name_or_path='bert-base-uncased', num_eval_steps=10, num_train_epochs=5, output_dir='out', seed=42, task_params={'ner': {'learning_rate': 3e-05, 'weight_decay': 0.01, 'adam_epsilon': 1e-08, 'max_grad_norm': 1.0, 'logging_steps': 500, 'save_steps': 500, 'output_mode': 'classification'}, 'cola': {'learning_rate': 3e-05, 'weight_decay': 0.01, 'adam_epsilon': 1e-08, 'max_grad_norm': 1.0, 'logging_steps': 500, 'save_steps': 500, 'output_mode': 'classification'}}, train_batch_size=32, warmup_proportion=0.1)
Some weights of the model checkpoint at bert-base-uncased were not used when initializing BertForTokenClassification: ['cls.predictions.bias', 'cls.predictions.transform.dense.weight', 'c

NER(10000) CoLA MRPC

In [ ]:
!bash run_cl.sh

流式输出内容被截断，只能显示最后 5000 行内容。
Iteration:  14% 1683/12272 [21:45<2:16:48,  1.29it/s]
Iteration:  14% 1684/12272 [21:46<2:17:08,  1.29it/s]
Iteration:  14% 1685/12272 [21:47<2:16:49,  1.29it/s]
Iteration:  14% 1686/12272 [21:48<2:16:53,  1.29it/s]
Iteration:  14% 1687/12272 [21:48<2:16:46,  1.29it/s]
Iteration:  14% 1688/12272 [21:49<2:16:56,  1.29it/s]
Iteration:  14% 1689/12272 [21:50<2:17:26,  1.28it/s]
Iteration:  14% 1690/12272 [21:51<2:17:00,  1.29it/s]
Iteration:  14% 1691/12272 [21:52<2:17:02,  1.29it/s]
Iteration:  14% 1692/12272 [21:52<2:17:21,  1.28it/s]
Iteration:  14% 1693/12272 [21:53<2:17:09,  1.29it/s]
Iteration:  14% 1694/12272 [21:54<2:17:12,  1.28it/s]
Iteration:  14% 1695/12272 [21:55<2:17:01,  1.29it/s]
Iteration:  14% 1696/12272 [21:55<2:16:32,  1.29it/s]
Iteration:  14% 1697/12272 [21:56<2:16:56,  1.29it/s]
Iteration:  14% 1698/12272 [21:57<2:16:34,  1.29it/s]
Iteration:  14% 1699/12272 [21:58<2:16:52,  1.29it/s]
Iteration:  14% 1700/12272 [21:59<2:17:14,  1.28it/s]
I

NER MRPC

In [ ]:
!bash run_cl.sh

2020-12-12 15:00:53.206770: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
Namespace(cuda=True, data_dir='data', device=device(type='cuda'), do_lower_case=True, eval_batch_size=32, eval_during_training=False, max_seq_length=128, model_name_or_path='bert-base-uncased', num_eval_steps=10, num_train_epochs=5, output_dir='out', seed=42, task_params={'ner': {'learning_rate': 3e-05, 'weight_decay': 0.01, 'adam_epsilon': 1e-08, 'max_grad_norm': 1.0, 'logging_steps': 500, 'save_steps': 500, 'output_mode': 'classification'}, 'mrpc': {'learning_rate': 3e-05, 'weight_decay': 0.01, 'adam_epsilon': 1e-08, 'max_grad_norm': 1.0, 'logging_steps': 500, 'save_steps': 500, 'output_mode': 'classification'}}, train_batch_size=32, warmup_proportion=0.1)
Some weights of the model checkpoint at bert-base-uncased were not used when initializing BertForTokenClassification: ['cls.predictions.bias', 'cls.predictions.transform.dense.weight', 'c

NER CoLA MRPC

In [ ]:
!bash run_cl.sh

  Cloning https://github.com/huggingface/transformers.git to /tmp/pip-req-build-1i4fxq6l
  Running command git clone -q https://github.com/huggingface/transformers.git /tmp/pip-req-build-1i4fxq6l
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
  Created wheel for transformers: filename=transformers-4.1.0.dev0-cp36-none-any.whl size=1448634 sha256=9548599893ec9ca056043d03b47046a871642f52f2acc57b6ec3b877f1bbf102
  Stored in directory: /tmp/pip-ephem-wheel-cache-eab3w9u8/wheels/33/eb/3b/4bf5dd835e865e472d4fc0754f35ac0edb08fe852e8f21655f
Successfully built transformers
--2020-12-12 11:33:50--  https://gist.githubusercontent.com/W4ngatang/60c2bdb54d156a41194446737ce03e2e/raw/17b8dd0d724281ed7c3b2aeeda662b92809aadd5/download_glue_data.py
Resolving gist.githubusercontent.com (gist.githubusercontent.com)... 151.101.0.133, 151.101.64.133, 151.101.128.133, ...
Connecting to gist.githubusercontent.com (gist.githubuserco

In [ ]:
!bash run_cl.sh

流式输出内容被截断，只能显示最后 5000 行内容。
Iteration:  60% 7382/12272 [48:12<31:59,  2.55it/s]
Iteration:  60% 7383/12272 [48:12<31:56,  2.55it/s]
Iteration:  60% 7384/12272 [48:13<32:03,  2.54it/s]
Iteration:  60% 7385/12272 [48:13<32:00,  2.55it/s]
Iteration:  60% 7386/12272 [48:14<31:57,  2.55it/s]
Iteration:  60% 7387/12272 [48:14<31:49,  2.56it/s]
Iteration:  60% 7388/12272 [48:14<31:55,  2.55it/s]
Iteration:  60% 7389/12272 [48:15<31:53,  2.55it/s]
Iteration:  60% 7390/12272 [48:15<31:52,  2.55it/s]
Iteration:  60% 7391/12272 [48:16<31:51,  2.55it/s]
Iteration:  60% 7392/12272 [48:16<32:00,  2.54it/s]
Iteration:  60% 7393/12272 [48:16<31:59,  2.54it/s]
Iteration:  60% 7394/12272 [48:17<31:58,  2.54it/s]
Iteration:  60% 7395/12272 [48:17<31:50,  2.55it/s]
Iteration:  60% 7396/12272 [48:18<31:50,  2.55it/s]
Iteration:  60% 7397/12272 [48:18<31:47,  2.56it/s]
Iteration:  60% 7398/12272 [48:18<31:45,  2.56it/s]
Iteration:  60% 7399/12272 [48:19<31:45,  2.56it/s]
Iteration:  60% 7400/12272 [48:19<31:

In [ ]:
!pip install git+https://github.com/huggingface/transformers.git

  Cloning https://github.com/huggingface/transformers.git to /tmp/pip-req-build-4u5luyhl
  Running command git clone -q https://github.com/huggingface/transformers.git /tmp/pip-req-build-4u5luyhl
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
     |████████████████████████████████| 890kB 14.2MB/s 
     |████████████████████████████████| 2.9MB 46.0MB/s 
  Created wheel for transformers: filename=transformers-4.1.0.dev0-cp36-none-any.whl size=1375102 sha256=a03342aafef8f6d370e3a51b2acb93f2306634300142ea0052a8bcc88b7a0897
  Stored in directory: /tmp/pip-ephem-wheel-cache-ozsvkxqd/wheels/33/eb/3b/4bf5dd835e865e472d4fc0754f35ac0edb08fe852e8f21655f
Successfully built transformers
  Created wheel for sacremoses: filename=sacremoses-0.0.43-cp36-none-any.whl size=893257 sha256=47d4c4f9d7cdac03f788cfacc7246b4548ec32f16408ecbe4d6aee88feacba97
  Stored in directory: /root/.cache/pip/wheels/29/3c/fd/7ce5c3f0666dab31a5012

In [ ]:
!pip install conllu

In [ ]:
!pip install seqeval

     |████████████████████████████████| 51kB 7.8MB/s 
  Created wheel for seqeval: filename=seqeval-1.2.2-cp36-none-any.whl size=16171 sha256=09d3680a7460c277a4af92fb5a29dd8abbd900c1814236907a1b225fd5510b70
  Stored in directory: /root/.cache/pip/wheels/52/df/1b/45d75646c37428f7e626214704a0e35bd3cfc32eda37e59e5f
Successfully built seqeval


In [ ]:
mkdir data/pos

mkdir: cannot create directory ‘data/pos’: No such file or directory


In [ ]:
!curl -L -o data/pos/dev.txt 'https://github.com/UniversalDependencies/UD_English-EWT/raw/master/en_ewt-ud-dev.conllu'
!curl -L -o data/pos/test.txt 'https://github.com/UniversalDependencies/UD_English-EWT/raw/master/en_ewt-ud-test.conllu'
!curl -L -o data/pos/train.txt 'https://github.com/UniversalDependencies/UD_English-EWT/raw/master/en_ewt-ud-train.conllu'

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   164  100   164    0     0    307      0 --:--:-- --:--:-- --:--:--   306
  0 1671k    0  1371    0     0   1425      0  0:20:00 --:--:--  0:20:00  1425
curl: (23) Failed writing body (0 != 1371)
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   165  100   165    0     0    336      0 --:--:-- --:--:-- --:--:--   336
  0 1672k    0  1371    0     0   1122      0  0:25:26  0:00:01  0:25:25  1122
curl: (23) Failed writing body (0 != 1371)
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   166  100   166    0     0    398      0 --:--:-- --:--:-- --:--:--   397
  0 12.8M    0     0    0     0      0      0

In [ ]:
rm -r data/cola/

rm: cannot remove 'data/cola/': No such file or directory


In [ ]:
rm -r data/mrpc/

rm: cannot remove 'data/mrpc/': No such file or directory


In [ ]:
!wget https://gist.githubusercontent.com/W4ngatang/60c2bdb54d156a41194446737ce03e2e/raw/17b8dd0d724281ed7c3b2aeeda662b92809aadd5/download_glue_data.py
!python download_glue_data.py --data_dir='data/' --tasks='CoLA'
!python download_glue_data.py --data_dir='data/' --tasks='MRPC'
!python download_glue_data.py --data_dir='data/' --tasks='RTE'

--2020-12-07 15:36:30--  https://gist.githubusercontent.com/W4ngatang/60c2bdb54d156a41194446737ce03e2e/raw/17b8dd0d724281ed7c3b2aeeda662b92809aadd5/download_glue_data.py
Resolving gist.githubusercontent.com (gist.githubusercontent.com)... 151.101.0.133, 151.101.64.133, 151.101.128.133, ...
Connecting to gist.githubusercontent.com (gist.githubusercontent.com)|151.101.0.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 8225 (8.0K) [text/plain]
Saving to: ‘download_glue_data.py’

download_glue_data. 100%[===================>]   8.03K  --.-KB/s    in 0s      

2020-12-07 15:36:30 (44.1 MB/s) - ‘download_glue_data.py’ saved [8225/8225]

	Completed!
Processing MRPC...
Local MRPC data not specified, downloading data from https://dl.fbaipublicfiles.com/senteval/senteval_data/msr_paraphrase_train.txt
	Completed!
	Completed!


In [ ]:
mv data/CoLA/ data/cola/

In [ ]:
mv data/MRPC/ data/mrpc/

In [ ]:
mv data/RTE/ data/rte/

In [ ]:
!python download_glue_data.py --data_dir='data/' --tasks='RTE'

	Completed!


In [ ]:
! /opt/bin/nvidia-smi

Thu Dec  3 23:43:43 2020       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 418.67       Driver Version: 418.67       CUDA Version: 10.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla T4            Off  | 00000000:00:04.0 Off |                    0 |
| N/A   40C    P8     9W /  70W |      0MiB / 15079MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU  

In [ ]:
!python run.py \
  --data_dir "data" \
  --task_params "cola.json" \
  --cuda \
  --do_lower_case \
  --model_name_or_path "bert-base-uncased" \
  --output_dir "out" \
  --train_batch_size 32 \
  --num_train_epochs 1 \
  --seed 42

In [ ]:
!python run.py \
  --data_dir "data" \
  --task_params "ner.json" \
  --cuda \
  --do_lower_case \
  --model_name_or_path "bert-base-uncased" \
  --output_dir "out" \
  --train_batch_size 32 \
  --num_train_epochs 1 \
  --seed 42

In [ ]:
!python run_ner.py \
  --data_dir "data/ner" \
  --do_lower_case \
  --model_type bert \
  --model_name_or_path "bert-base-uncased" \
  --output_dir "output" \
  --per_gpu_train_batch_size 32 \
  --num_train_epochs 1 \
  --seed 42\
  --do_train \
  --do_eval \
  --do_predict

python3: can't open file 'run_ner.py': [Errno 2] No such file or directory


In [ ]:
rm -r output2/

In [ ]:
!pip install transformers==2.8.0

In [ ]:
!bash run_cl.sh

--2020-12-06 17:29:10--  https://gist.githubusercontent.com/W4ngatang/60c2bdb54d156a41194446737ce03e2e/raw/17b8dd0d724281ed7c3b2aeeda662b92809aadd5/download_glue_data.py
Resolving gist.githubusercontent.com (gist.githubusercontent.com)... 151.101.0.133, 151.101.64.133, 151.101.128.133, ...
Connecting to gist.githubusercontent.com (gist.githubusercontent.com)|151.101.0.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 8225 (8.0K) [text/plain]
Saving to: ‘download_glue_data.py.1’

download_glue_data. 100%[===================>]   8.03K  --.-KB/s    in 0s      

2020-12-06 17:29:11 (88.1 MB/s) - ‘download_glue_data.py.1’ saved [8225/8225]

***********************
Namespace(data_dir='data/', path_to_mrpc='', tasks='CoLA,MRPC,RTE')
	Completed!
Processing MRPC...
Local MRPC data not specified, downloading data from https://dl.fbaipublicfiles.com/senteval/senteval_data/msr_paraphrase_train.txt
	Completed!
	Completed!
2020-12-06 17:29:14.823715: I tensorflow/stream_e

In [ ]:
!python run.py \
  --data_dir "data" \
  --task_params "order1.json" \
  --cuda \
  --do_lower_case \
  --model_name_or_path "bert-base-uncased" \
  --output_dir "out" \
  --train_batch_size 32 \
  --num_train_epochs 5 \
  --seed 42

2020-12-04 14:20:51.366738: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
Namespace(cuda=True, data_dir='data', device=device(type='cuda'), do_lower_case=True, eval_batch_size=32, eval_during_training=False, max_seq_length=128, model_name_or_path='bert-base-uncased', num_eval_steps=10, num_train_epochs=5, output_dir='out', seed=42, task_params={'cola': {'learning_rate': 3e-05, 'weight_decay': 0.01, 'adam_epsilon': 1e-08, 'max_grad_norm': 1.0, 'logging_steps': 500, 'save_steps': 500, 'output_mode': 'classification'}, 'mrpc': {'learning_rate': 3e-05, 'weight_decay': 0.01, 'adam_epsilon': 1e-08, 'max_grad_norm': 1.0, 'logging_steps': 500, 'save_steps': 500, 'output_mode': 'classification'}, 'rte': {'learning_rate': 3e-05, 'weight_decay': 0.01, 'adam_epsilon': 1e-08, 'max_grad_norm': 1.0, 'logging_steps': 500, 'save_steps': 500, 'output_mode': 'classification'}}, train_batch_size=32, warmup_proportion=0.1)
12/04/2020 1

In [ ]:
rm -r out/

In [ ]:
!python run.py \
  --data_dir "data" \
  --task_params "order1.json" \
  --cuda \
  --do_lower_case \
  --model_name_or_path "bert-base-uncased" \
  --output_dir "out" \
  --train_batch_size 32 \
  --num_train_epochs 1 \
  --seed 42

2020-12-02 22:17:20.937496: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
Namespace(cuda=True, data_dir='data', device=device(type='cuda'), do_lower_case=True, eval_batch_size=32, eval_during_training=False, max_seq_length=128, model_name_or_path='bert-base-uncased', num_eval_steps=10, num_train_epochs=1, output_dir='out', seed=42, task_params={'cola': {'learning_rate': 3e-05, 'weight_decay': 0.01, 'adam_epsilon': 1e-08, 'max_grad_norm': 1.0, 'logging_steps': 500, 'save_steps': 500, 'output_mode': 'classification'}, 'mrpc': {'learning_rate': 3e-05, 'weight_decay': 0.01, 'adam_epsilon': 1e-08, 'max_grad_norm': 1.0, 'logging_steps': 500, 'save_steps': 500, 'output_mode': 'classification'}, 'rte': {'learning_rate': 3e-05, 'weight_decay': 0.01, 'adam_epsilon': 1e-08, 'max_grad_norm': 1.0, 'logging_steps': 500, 'save_steps': 500, 'output_mode': 'classification'}}, train_batch_size=32, warmup_proportion=0.1)
12/02/2020 2

In [ ]:
a = 1
print(a)

1


In [ ]:
!python run.py \
  --data_dir "data" \
  --task_params "pos.json" \
  --cuda \
  --do_lower_case \
  --model_name_or_path "bert-base-uncased" \
  --output_dir "out" \
  --train_batch_size 32 \
  --num_train_epochs 1 \
  --seed 42

2020-12-02 17:38:25.849805: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
Namespace(cuda=True, data_dir='data', device=device(type='cuda'), do_lower_case=True, eval_batch_size=32, eval_during_training=False, max_seq_length=128, model_name_or_path='bert-base-uncased', num_eval_steps=10, num_train_epochs=1, output_dir='out', seed=42, task_params={'pos': {'learning_rate': 3e-05, 'weight_decay': 0.01, 'adam_epsilon': 1e-08, 'max_grad_norm': 1.0, 'logging_steps': 500, 'save_steps': 500, 'output_mode': 'classification'}}, train_batch_size=32, warmup_proportion=0.1)
12/02/2020 17:38:27 - INFO - filelock - Lock 140487265803960 acquired on /root/.cache/torch/transformers/4dad0251492946e18ac39290fcfe91b89d370fee250efe9521476438fe8ca185.7156163d5fdc189c3016baca0775ffce230789d7fa2a42ef516483e4ca884517.lock
12/02/2020 17:38:27 - INFO - transformers.file_utils - https://s3.amazonaws.com/models.huggingface.co/bert/bert-base-uncas

In [ ]:
!python main.py \
  --data_dir "data" \
  --task_params "example_task.json" \
  --cuda \
  --tokenizer_name "bert" \
  --do_lower_case \
  --model_type "bert-base-uncased" \
  --output_dir "out" \
  --per_gpu_batch_size 32 \
  --num_train_epochs 2 \
  --seed 42 

2020-12-02 22:52:07.293859: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
Namespace(cuda=True, data_dir='data', device=device(type='cuda'), do_lower_case=True, max_seq_length=128, model_type='bert-base-uncased', n_gpu=1, num_train_epochs=2, output_dir='out', per_gpu_batch_size=32, plot=False, seed=42, task_params={'cola': {'learning_rate': 3e-05, 'weight_decay': 0.0, 'adam_epsilon': 1e-08, 'max_grad_norm': 1.0, 'logging_steps': 500, 'save_steps': 500, 'output_mode': 'classification'}, 'mrpc': {'learning_rate': 3e-05, 'weight_decay': 0.0, 'adam_epsilon': 1e-08, 'max_grad_norm': 1.0, 'logging_steps': 500, 'save_steps': 500, 'output_mode': 'classification'}, 'rte': {'learning_rate': 3e-05, 'weight_decay': 0.0, 'adam_epsilon': 1e-08, 'max_grad_norm': 1.0, 'logging_steps': 500, 'save_steps': 500, 'output_mode': 'classification'}}, tokenizer_name='bert', warmup_steps=0)
{}
{'cola': <transformers.data.processors.glue.Cola